src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -210,19 +214,53 @@ public boolean isValidIdentifierForTerm(SQLIdentifierExpr expr) {
             );
     }
 
-    public boolean hasIdenticalMappings(TermFieldScope scope, Map<String, String> indexToType) {
+    public void checkMappingCompatibility(TermFieldScope scope, Map<String, String> indexToType) {
         if (scope.getMapper().isEmpty()) {
             throw new VerificationException("Unknown index " + indexToType.keySet());
         }
+        // Collect all FieldMappings into hash set and ignore index/type names. set.size > 1 means FieldMappings NOT unique,
+        // needs further compatibility check
+        Set<FieldMappings> set = curScope().getMapper().allMappings().stream().
+                                 flatMap(typeMappings -> typeMappings.allMappings().stream()).
+                                 collect(Collectors.toSet());
 
-        if (isMappingOfAllIndicesDifferent()) {
-            return false;
-        }
+        FieldMappings fieldMappings = null;
+
+        if (set.size() > 1) {
+            Map<String, Map<String, Object>> map = new LinkedHashMap<>();
```

Comment:
please use more descriptive names for variables.

Reply:
seems you are not using any of the LinkedList features of this map, so just make it a `HashMap`

Reply:
> seems you are not using any of the LinkedList features of this map, so just make it a `HashMap`

I created this as LinkedHashMap because all the fieldMappings we generated elsewhere are created this way, and I think it would be better to keep it the same data type to avoid mismatch issue.

Reply:
the fact is you use only `put` and `contains`, and for that you don't need any ordering (which is what `LinkedHashMap` adds to `HashMap`)

Reply:
That makes sense. Will change to hashMap